File: trunk/ShadowFrameworkJS/SF20Core/src/shadow/pipeline/SFStructureData.js

function SFStructureData(structure, genericsParameters){
		this.structure=structure;
	values=new SFValuenf[structure.size()];//Warning: Not well Identified 
	int index=0;//Warning: Not well Identified 
}
function SFStructureData(structure){
		this.structure=structure;
	values=new SFValuenf[structure.size()];//Warning: Not well Identified 
	int index=0;//Warning: Not well Identified 
}

SFStructureData.prototype = {

	size:function(){
	return values.length;//Warning: Not well Identified 
	},

	getStructure:function(){
		return this.structure;
	},

	getValue:function(index){
		return this.values[index];
	},

	getValues:function(){
		return this.values;
	}

};

File: trunk/SFDataApplicationTestC++/shadow/pipeline/SFStructureData.cpp
//
//  SFStructureData.cpp
//  
//
//  Created by Alessandro Martinelli on 18/10/12.
//  Copyright (c) 2012 Alessandro Martinelli. All rights reserved.
//

#include "SFStructureData.h"

namespace sf{

SFStructureData::SFStructureData(SFPipelineStructure* structure) {
    this->structure=structure;
    short length=structure->size();
    SFParameter* parameters=structure->getAllParameters();
    values=new SFValue*[length];
    //cout << "values size "<< structure->size()<<endl;
    
    valuesSize=structure->size();
    
    for(int i=0;i<length;i++){
        values[i]=SFParameteri::generateValue(parameters[i].getType());
    }
}

int SFStructureData::size() {
    return valuesSize;
}

SFPipelineStructure* SFStructureData::getStructure() {
    return structure;
}

SFValue* SFStructureData::getValue(int index) {
    return values[index];
}

SFValue** SFStructureData::getValues() {
    return values;
}

}
